collection-scripts/gather_performance_profile_controller

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+
+function ppc_nodes(){
+  
+check_node_gather_pods_ready() {
+   line=$(oc get ds perf-node-gather-daemonset -o=custom-columns=DESIRED:.status.desiredNumberScheduled,READY:.status.numberReady --no-headers -n perf-node-gather)
+
+    IFS=$' '
+    read desired ready <<< $line
+    IFS=$'\n'
+
+    if [[ "$desired" != "0" ]] && [[ "$ready" == "$desired" ]]
+    then
+       return 0
+    else
+       return 1
+    fi
+}
+
+IFS=$'\n'
+
+BASE_COLLECTION_PATH="/must-gather"
+NODES_PATH=${BASE_COLLECTION_PATH}/nodes
+mkdir -p ${NODES_PATH}
+NAMESPACE_MANIFEST="/etc/performance-profile-node-gather/namespace.yaml"
+SERVICEACCOUNT_MANIFEST="/etc/performance-profile-node-gather/serviceaccount.yaml"
+DAEMONSET_MANIFEST="/etc/performance-profile-node-gather/daemonset.yaml"
+CLUSTER_ROLE_MANIFEST="/etc/performance-profile-node-gather/clusterrole.yaml"
+CLUSTER_ROLE_BINDING_MANIFEST="/etc/performance-profile-node-gather/clusterrolebinding.yaml"
+NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+# Once you start the pod, the Kubernetes will set the pod hostname to the name of the pod
+# https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields
+POD_NAME=${HOSTNAME}
+MUST_GATHER_IMAGE=$(oc get pod -n $NAMESPACE $POD_NAME -o jsonpath="{.spec.containers[0].image}")
+
+POD_IP=$(hostname -I |  tr -d "[:blank:]" )
+echo "[$NAMESPACE/$POD_IP/$POD_NAME]" >> ${BASE_COLLECTION_PATH}/debug
+oc get pod -n $NAMESPACE $POD_NAME -o json >> ${BASE_COLLECTION_PATH}/debug
+
+sed -i -e "s#MUST_GATHER_IMAGE#$MUST_GATHER_IMAGE#" $DAEMONSET_MANIFEST
+
+oc create -f $NAMESPACE_MANIFEST
```

Comment:
Can we 'reuse' the exsisting must-gather namespace; instead of creating a new namespace? 

By putting all the collection artifacts in the 'must-gather' namespace; when the `oc` command that invokes must-gather finishes; a clean up operation is done (removing all of the pods, and other artifacts that we create). 

By spawning off new namespaces; it's possible that we could leave around lingering artifacts.

Reply:
I am facing this issue if I try to reuse same namespace:
  Warning  FailedCreate  33s (x15 over 115s)  daemonset-controller  Error creating: pods "perf-node-gather-daemonset-" is forbidden: autoscaling.openshift.io/ManagementCPUsOverride the pod namespace "openshift-must-gather-f2s49" does not allow the workload type management